google/cloud/bigquery/_job_helpers.py

```diff
@@ -497,9 +517,20 @@ def query_and_wait(
         request_body["jobCreationMode"] = client.default_job_creation_mode
 
     def do_query():
-        request_body["requestId"] = make_job_id()
+        request_id = make_job_id()
+        request_body["requestId"] = request_id
         span_attributes = {"path": path}
 
+        callback(
+            QuerySentEvent(
```

Comment:
Same here. I think it might be useful to distinguish query retries as a separate event.